config/config_test.go

```diff
@@ -70,6 +71,87 @@ func TestDefaultSampler(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestServiceNameFromEnv(t *testing.T) {
+	os.Setenv(envServiceName, "my-service")
+
+	_, c, err := FromEnv().New("")
```

Comment:
I don't quite like that callers are required to pass an empty string if they expect the service name to be specified via env var, but perhaps it's acceptable?

Reply:
as the service name is a member of configuration, we could deprecate `New` and create `newTracer` without requiring serviceName.

Reply:
oh you had the same suggestion here https://github.com/jaegertracing/jaeger-client-go/pull/275#discussion_r177690878

I like it